src/SDK/SdkAutoloader.php

```diff
@@ -30,80 +36,121 @@ public static function autoload(): bool
+    {
+        $propagator = (new PropagatorFactory())->create();
+        if (Sdk::isDisabled()) {
+            //@see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#general-sdk-configuration
+            return $configurator->withPropagator($propagator);
+        }
+        $emitMetrics = Configuration::getBoolean(Variables::OTEL_PHP_INTERNAL_METRICS_ENABLED);
+
+        $resource = ResourceInfoFactory::defaultResource();
+        $exporter = (new ExporterFactory())->create();
+        $meterProvider = (new MeterProviderFactory())->create($resource);
+        $spanProcessor = (new SpanProcessorFactory())->create($exporter, $emitMetrics ? $meterProvider : null);
+        $tracerProvider = (new TracerProviderBuilder())
+            ->addSpanProcessor($spanProcessor)
+            ->setResource($resource)
+            ->setSampler((new SamplerFactory())->create())
+            ->build();
+
+        $loggerProvider = (new LoggerProviderFactory())->create($emitMetrics ? $meterProvider : null, $resource);
+        $eventLoggerProvider = (new EventLoggerProviderFactory())->create($loggerProvider);
+
+        ShutdownHandler::register($tracerProvider->shutdown(...));
+        ShutdownHandler::register($meterProvider->shutdown(...));
+        ShutdownHandler::register($loggerProvider->shutdown(...));
+
+        return $configurator
+            ->withTracerProvider($tracerProvider)
+            ->withMeterProvider($meterProvider)
+            ->withLoggerProvider($loggerProvider)
+            ->withEventLoggerProvider($eventLoggerProvider)
+            ->withPropagator($propagator)
+        ;
+    }
+
+    private static function createSdkFromFileConfig(): Sdk
+    {
+        $file = Configuration::getString(Variables::OTEL_EXPERIMENTAL_CONFIG_FILE);
+        $config = SdkConfiguration::parseFile($file);
+
+        return $config
+            ->create()
+            ->setAutoShutdown(true)
+            ->build();
+    }
+
     /**
-     * Test whether a request URI is set, and if it matches the excluded urls configuration option
+     * Register any auto-instrumentation configured through SPI
      *
-     * @internal
+     * @phan-suppress PhanUndeclaredClassMethod
      */
-    public static function isIgnoredUrl(): bool
+    private static function registerInstrumentations(?Sdk $sdk): void
     {
-        $ignoreUrls = Configuration::getList(Variables::OTEL_PHP_EXCLUDED_URLS, []);
-        if ($ignoreUrls === []) {
-            return false;
+        if (!Configuration::has(Variables::OTEL_PHP_INSTRUMENTATION_CONFIG_FILE)) {
+            return;
         }
-        $url = $_SERVER['REQUEST_URI'] ?? null;
-        if (!$url) {
-            return false;
+        $file = Configuration::getString(Variables::OTEL_PHP_INSTRUMENTATION_CONFIG_FILE);
+        $configuration = Instrumentation::parseFile($file)->create();
+        $storage = Context::storage();
```

Comment:
We cannot use `Globals::storage()` here. `initialize_fiber_handlers.php` as well as users of the swoole context storage might `Context::setStorage()` a different context storage during startup.

Either we drop the `$storage` parameter from `Instrumentation::register()` or we inject a context storage that delegates to the global storage.

Reply:
Dropped storage from `Instrumentation::register()`